LoopKit/LoopAlgorithm/LoopAlgorithmInput.swift

```diff
@@ -7,15 +7,347 @@
 //
 
 import Foundation
+import HealthKit
 
-public enum DoseRecommendationType: String {
-    case manualBolus
-    case automaticBolus
-    case tempBasal
+public enum AlgorithmInputDecodingError: Error {
+    case invalidDoseRecommendationType
+    case invalidInsulinType
+    case doseRateMissing
+    case doseVolumeMissing
 }
 
 public struct LoopAlgorithmInput {
-    public var predictionInput: LoopPredictionInput
-    public var predictionDate: Date
-    public var doseRecommendationType: DoseRecommendationType
+    public var predictionStart: Date
+    public var glucoseHistory: [StoredGlucoseSample]
+    public var doses: [DoseEntry]
+    public var carbEntries: [StoredCarbEntry]
+    public var basal: [AbsoluteScheduleValue<Double>]
+    public var sensitivity: [AbsoluteScheduleValue<HKQuantity>]
+    public var carbRatio: [AbsoluteScheduleValue<Double>]
+    public var target: GlucoseRangeTimeline
+    public var suspendThreshold: HKQuantity?
```

Comment:
We can look at the "interface" for the loop algorithm at a couple of different levels.  At the file format level, everything has been simplified to the bare minimum of fields, and any glucose related fields are specified as simple doubles, assuming mg/dL. 

This input struct (and the algorithm itself) still uses HKQuantity instances for glucose related values. I do think it would be nice to move the algorithm in the direction of requiring the bare minimum input types.  Currently not only does it require healthkit, but the dose/carb/glucose objects have a lot of fields/data that are not strictly necessary for generating a recommendation.

I've making progress in chunks.  The first chunk here is extracting the algo from its dependency on data stores and other async calls, supporting timeline based settings (instead of daily schedules), and creating file formats to represent the entire input and output in a simplified manner.  Probably the next chunk would be to tackle the input types; making them not dependent on HK, and eliminating unneeded fields, and perhaps representing them as protocols. Maybe also extract all the related code to a LoopAlgorithmKit.